src/core/renderer/MCHTMLRenderer.cpp

```diff
@@ -464,8 +464,9 @@ static String * htmlForAbstractMultipartMixed(AbstractMultipart * part, htmlRend
         AbstractPart * subpart = (AbstractPart *) part->parts()->objectAtIndex(i);
         String * substring = htmlForAbstractPart(subpart, context);
         if (context->pass != 0) {
-            if (substring == NULL)
-                return NULL;
+            if (substring == NULL) {
```

Comment:
If you're getting null here, it indicates a failure in `htmlForAbstractPart`.  I don't think this should be the workaround for that kind of logic error.